src/trading/base.py

```
@@ -8,13 +12,79 @@
+            "tx_signature": self.tx_signature,
+            "error_message": self.error_message,
+            "amount": self.amount,
+            "price": self.price,
+        }
+
+
+class Trader(ABC):
+    """Enhanced base interface for trading operations with platform support."""
+
+    @abstractmethod
+    async def execute(self, token_info: TokenInfo, *args, **kwargs) -> TradeResult:
```

Ruff (0.12.2)

Missing type annotation for `*args`

(ANN002)